gql/resolver.go

```diff
@@ -134,7 +142,22 @@ func (r *resolver) Deals(ctx context.Context, args dealsArgs) (*dealListResolver
 	if args.Query.Set && args.Query.Value != nil {
 		query = *args.Query.Value
 	}
-	deals, count, more, err := r.dealList(ctx, query, args.Cursor, offset, limit)
+
+	filter := map[string]interface{}{}
+	if args.Filter.Checkpoint.String != "" {
+		filter["Checkpoint"] = args.Filter.Checkpoint.String
+	}
+	if args.Filter.IsOffline.Set && args.Filter.IsOffline.Value != nil {
+		filter["IsOffline"] = args.Filter.IsOffline.Value
+	}
+	if args.Filter.TransferType.Set && args.Filter.TransferType.Value != nil {
+		filter["TransferType"] = args.Filter.TransferType.Value
+	}
+	if args.Filter.VerifiedDeal.Set && args.Filter.VerifiedDeal.Value != nil {
+		filter["VerifiedDeal"] = args.Filter.VerifiedDeal.Value
+	}
```

Comment:
I think the code may end up simpler if we add a FilterOptions struct as a parameter to DealList at the db layer that looks similar to filterArgs (but where each field is a pointer so that it can be nil). That way we also don't expose the db fields at the resolver layer.